src/ui/src/builder/settings/BuilderFieldsToolsWithMcp.vue

```diff
@@ -0,0 +1,494 @@
+import injectionKeys from "@/injectionKeys";
+import { useWriterApi } from "@/composables/useWriterApi";
+import type { WriterApiMcpTool } from "@/writerApi";
+
+type FunctionTool = {
+	type: "function";
+	description: string;
+	parameters: Record<
+		string,
+		{
+			type: string;
+			description: string;
+		}
+	>;
+};
+
+type GraphTool = {
+	type: "graph";
+	graph_ids: string[];
+};
+
+type WebSearchTool = {
+	type: "web_search";
+	include_domains?: string[];
+	exclude_domains?: string[];
+	include_raw_content?: boolean;
+};
+
+type McpToolFunction = {
+	description?: string;
+	parameters?: Record<string, unknown>;
+	name?: string;
+	[key: string]: unknown;
+};
+
+type McpToolType = {
+	type: "mcp";
+	appId: string;
+	functionName: string;
+	function: McpToolFunction;
+};
+
+type Tool = FunctionTool | GraphTool | WebSearchTool | McpToolType;
+
+const wf = inject(injectionKeys.core);
+
+const { writerApi } = useWriterApi();
+
+const mcpToolsData = ref<WriterApiMcpTool[]>([]);
+const mcpToolsLoading = ref(false);
+const mcpToolsError = ref<Error | undefined>(undefined);
+const expandedApps = ref<Set<string>>(new Set());
+
+type GroupedTool = {
+	functions: WriterApiMcpTool[];
+	logo?: string;
+};
+
+const groupedTools = computed(() => {
+	const grouped: Record<string, GroupedTool> = {};
+
+	for (const tool of mcpToolsData.value) {
+		const appName = tool.appName || "Unknown App";
+		if (!grouped[appName]) {
+			grouped[appName] = {
+				functions: [],
+				logo: tool.connector?.logo,
+			};
+		}
+		grouped[appName].functions.push(tool);
+	}
+	return grouped;
+});
+
+const selectedCountsByApp = computed(() => {
+	const counts: Record<string, number> = {};
+	for (const [appName, group] of Object.entries(groupedTools.value)) {
+		counts[appName] = group.functions.filter((tool) =>
+			isToolSelected(tool),
+		).length;
+	}
+	return counts;
+});
+
+function toggleApp(appName: string) {
+	if (expandedApps.value.has(appName)) {
+		expandedApps.value.delete(appName);
+	} else {
+		expandedApps.value.add(appName);
+	}
+}
+
+function normalizeToolName(tool: WriterApiMcpTool): string {
+	return `${tool.appName}_${tool.functionName}`.replace(
+		/[^a-zA-Z0-9_]/g,
+		"_",
+	);
+}
+
+function isToolSelected(tool: WriterApiMcpTool): boolean {
+	return normalizeToolName(tool) in tools.value;
+}
+
+function toggleTool(tool: WriterApiMcpTool, checked: boolean) {
+	const toolName = normalizeToolName(tool);
+
+	if (checked) {
+		const mcpTool: McpToolType = {
+			type: "mcp",
+			appId: tool.appId,
+			functionName: tool.functionName,
+			function: tool.function,
+		};
+		fieldViewModel.value = JSON.stringify({
+			...tools.value,
+			[toolName]: mcpTool,
+		});
+	} else {
+		const updatedTools = { ...tools.value };
+		delete updatedTools[toolName];
+		fieldViewModel.value = JSON.stringify(updatedTools);
+	}
+}
+
+async function loadMcpTools() {
+	const orgId = wf.writerOrgId.value || 1;
+
+	if (!orgId) {
```

Comment:
The `if` is not reachable
```suggestion
	const orgId = wf.writerOrgId.value;

	if (!orgId) {
```